### PR TITLE
First draft of CONTRIBUTING

### PR DESCRIPTION
**Pull request by thomwiggers**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Contributing new schemes to PQClean
```
Comment (henrydcase) [inline CONTRIBUTING.md]: Cool! Does it mean, it's OK to submit SIDH now?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+For a signature scheme, these steps are equivalent, but the API is slightly different.
+See the section [API](#API) below.
+
+1. Fork our repository. You will be creating a pull request soon.
+    * **Tip:** Do not wait until you think you have gotten everything perfect, before you open the pull request.
+      We set up things so Github and the CI environment will give you feedback and guidance on the steps to follow.
+
+2. Create the following folder structure: `crypto_kem/yourschemename/clean`.
+We follow the SUPERCOP layout, so please create a separate folder for each parameter set.
+For now, we only accept **pure, portable C code**
```
Comment (henrydcase) [inline CONTRIBUTING.md]: C99?

File: CONTRIBUTING.md (hunk 1 above)
Comment (thomwiggers) [reply CONTRIBUTING.md]: There are some things still moving; but we hope that we are in a shape to accept submissions soon. Notably, I also need to solve #1

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+### Signature schemes
+
+Functions:
+
+```c
+int PQCLEAN_YOURSCHEME_CLEAN_crypto_sign_keypair(unsigned char *pk, unsigned char *sk);
+int PQCLEAN_YOURSCHEME_CLEAN_crypto_sign(unsigned char *sm, unsigned long long *smlen, const unsigned char *msg, unsigned long long len, const unsigned char *sk);
+int PQCLEAN_YOURSCHEME_CLEAN_crypto_sign_open(unsigned char *m, unsigned long long *mlen, const unsigned char *sm, unsigned long long smlen, const unsigned char *pk);
+```
```
Comment (dstebila) [inline CONTRIBUTING.md]: What about the `crypto_sign_signature` and `crypto_sign_verify` forms for signature schemes?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+### Return codes
+
+Your schemes should return 0 on success, or a negative value on failure.
+Notably, `crypto_sign_open` should return `-1` if signature verification failed.
+
+Contributing to the framework of PQClean
+========================================
+
+We also welcome contributions to the testing framework. Open an issue or pull request on Github and we will review your suggestion. In general, we are always looking to improve the experience of submitters of schemes and of people consuming the implementations collected by this project. Please do bear in mind the intentions of this project.
```
Comment (dstebila) [inline CONTRIBUTING.md]: I don't know what the last sentence is trying to say.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+Contributing new schemes to PQClean
+===================================
+
+Why contribute to PQClean
+-------------------------
+
+PQClean hopes to provide your scheme to people who want to integrate post-quantum cryptography into their own libraries and applications. But our extensive testing framework might also help you catch bugs in your implementation, that might have otherwise gone unnoticed. We run our builds on (emulated) ARMv7, ARMv8, x86 and amd64. Also, we apply static and dynamic analysis tools.
```
Comment (dstebila) [inline CONTRIBUTING.md]: And PowerPC?

File: CONTRIBUTING.md (hunk 4 above)
Comment (thomwiggers) [reply CONTRIBUTING.md]: Don't propose taking over `liboqs`'s job

File: CONTRIBUTING.md (hunk 3 above)
Comment (thomwiggers) [reply CONTRIBUTING.md]: Dilithium does not define these methods...

File: CONTRIBUTING.md (hunk 4 above)
Comment (thomwiggers) [reply CONTRIBUTING.md]: Removed this line.

File: CONTRIBUTING.md (hunk 5 above)
Comment (thomwiggers) [reply CONTRIBUTING.md]: Good point, added.

**Comment by thomwiggers** [comment]
I've rebased this branch, for those who have local copies.

File: CONTRIBUTING.md (hunk 3 above)
Comment (dstebila) [reply CONTRIBUTING.md]: So Dilithium does not meet the requirements stated in our README.  :)

**Comment by thomwiggers** [comment]
Follow-up question: how do our tests not catch this 🤔

**Pull request merged**